src/reporter.ts

```diff
@@ -449,38 +451,66 @@ export default class Reporter {
         });
       });
 
-      if (this.opts.artifactUploadDir) {
-        const artifactPath = path.join(this.opts.artifactUploadDir, specName);
-        if (fs.existsSync(artifactPath)) {
-          const entries = await readdir(path.resolve(artifactPath), {
-            withFileTypes: true,
-          });
-
-          for (const entry of entries) {
-            if (!entry.isFile()) {
-              continue;
-            }
-
-            const entryPath = path.join(artifactPath, entry.name);
-            assets.push({
-              filename: entry.name,
-              path: entryPath,
-              data: fs.createReadStream(path.resolve(entryPath)),
-            });
-          }
-        }
+      const artifacts = await this.collectArtifacts(specName);
+      assets.push(...artifacts);
+    }
+
+    return assets;
+  }
+
+  /**
+   * Collects assets for the console log and sauce-test-report.json.
+   *
+   * @param result - The result of the test run to generate the console log.
+   * @param report - The sauce test report to be attached as an asset.
+   * @returns An array of assets containing the console log and sauce test report.
+   */
+  collectLogsAndReport(result: RunResult, report: TestRun): Asset[] {
```

Comment:
💡 Would it be perhaps better to have two distinct methods? This method doesn't seem to be able to stand the test of time. If we were to have a need for yet another asset, would the method be called `collectLogsAndReportAndXyz` and so on and so forth?

`getConsoleLog` and `getSauceTestReport` may be preferable and are self explanatory wherever they are end up being used.

Reply:
Yeah. Better!